queries/nim/highlights.scm

```diff
@@ -0,0 +1,797 @@
+
+; =============================================================================
+; @float                ; floating-point number literals
+
+(float_literal) @float
+
+; =============================================================================
+; @function         ; function definitions
+
+(proc_declaration
+  name: [
+    (identifier) @function
+    (accent_quoted (identifier) @function)
+    (exported_symbol (identifier) @function)
+    (exported_symbol (accent_quoted (identifier) @function))
+  ])
+
+(func_declaration
+  name: [
+    (identifier) @function
+    (accent_quoted (identifier) @function)
+    (exported_symbol (identifier) @function)
+    (exported_symbol (accent_quoted (identifier) @function))
+  ])
+
+(iterator_declaration
+  name: [
+    (identifier) @function
+    (accent_quoted (identifier) @function)
+    (exported_symbol (identifier) @function)
+    (exported_symbol (accent_quoted (identifier) @function))
+  ])
+
+(converter_declaration
+  name: [
+    (identifier) @function
+    (accent_quoted (identifier) @function)
+    (exported_symbol (identifier) @function)
+    (exported_symbol (accent_quoted (identifier) @function))
+  ])
+
+; =============================================================================
+; @function.call    ; function calls
+
+(call
+  function: [
+    (identifier) @function.call
+    (accent_quoted (identifier) @function.call)
+    ; generic types
+    (bracket_expression left: (identifier) @function.call)
+    (bracket_expression left: (accent_quoted (identifier) @function.call))
+    ; dot accessor
+    (dot_expression right: (identifier) @function.call)
+    (dot_expression right: (accent_quoted (identifier) @function.call))
+    ; both
+    (bracket_expression left:
+      (dot_expression right: (identifier) @function.call))
+    (bracket_expression left:
+      (dot_expression right: (accent_quoted (identifier) @function.call)))
+  ])
+
+; generalized_string is a function call
+; `identifier"string literal"`
+; is short for
+; `identifier(r"string literal")`
+(generalized_string 
+  function: [
+    (identifier) @function.call
+    (accent_quoted (identifier) @function.call)
+  ])
+
+; call with leading literal
+(dot_expression
+  left: [
+    (nil_literal)
+    (integer_literal)
+    (float_literal)
+    (custom_numeric_literal)
+    (char_literal)
+    (interpreted_string_literal)
+    (long_string_literal)
+    (raw_string_literal)
+    (generalized_string)
+    (array_construction)
+    ; for sequences
+    (prefix_expression 
+      operator: (operator) @_at
+      (array_construction)
+      (#eq? @_at "@"))
+    (tuple_construction)
+    (curly_construction)
+  ]
+  right: [
+    (identifier) @function.call
+    (accent_quoted (identifier) @function.call)
+  ])
+; NOTE: will double capture as @function.call if it also has argument_list
+
+; =============================================================================
+; @function.builtin ; built-in functions
+; unused
```

Comment:
What about [these builtins](https://nim-lang.org/docs/manual.html#procedures-nonoverloadable-builtins)?

Reply:
nice. I never found that reference. I was arguing with me which builtins (from system.nim) to include and didn't know where to draw the line so I left it completely.

Reply:
Honestly I don't like this list. Put aside that it seems to be out of date, since `definedInScope` doesn't even exist apparently.

Like I said, there are just [sooo many builtin functions](https://nim-lang.org/docs/system.html#12) that could deserve to be captured here.
Eg, why should `sizeof` be in it, but not `typeof`?

Any line drawn would be arbitrary and the toll on the performance would grow and grow with the list of identifiers to check for.

Reply:
same goes for `@type.builtin` btw. With that one it would be also much more impactful on performance, because I would need to duplicate all the `@type` queries to check for all the possible identifiers.